First restructuring pass for auth section

Oglopf opened the pull request:
**Modify this link to include the branch name, and possibly the page this PR modifies**:

https://osc.github.io/ood-documentation-test/auth-page-restruct-1035/

**Add your description here**
First pass at restructuring some of the auth page to help with the flow and emphasis on parts that _must_ be done which users have a history of missing (like user mapping).

Part of issue #1035

genericdata committed:
Remove 20.04 from supported OS (#1112)

Remove 20.04 from various pages mentioning it.

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

osc-bot committed:
Merge branch 'latest' into develop

treydock committed:
Document the need to import GPG key (#1137)

genericdata committed:
Add popup_title to path_selector (#1143)

Oglopf committed:
First restructuring pass for auth section

moffatsadeghi commented:
ToC is the last piece to tie up - since hitting some funkiness with the rendering.  Discussed and walked through each change.  Looks fantastic on first walkthrough!

Oglopf committed:
major movement of some data and toc for authentication for clarity and easier flow

Oglopf commented:
I think I have the ToC finally working correctly and all links seem to be correct. I'm still clicking through some pages to check I haven't missed anything yet.

This makes the landing page for the auth very lean but it is clear what you need to do and i removed a lot of redundancy in wording and warnings and merged some things. The idea is the page is incredibly streamlined so that it is hard to miss what is required, and it's now designed to let you "fall through" the docs section of the ToC then spit you back out to a "Known Auth Solutions" page where the user can select their auth, rather than just being dumped right into OIDC as previously.

I did some serious rework on the overview, to the point i removed the file and even the old authentication page, split the authentication between the overview and the solutions, then folded the overview into that new ood-auth page as it felt like 3 pages all said the same thing in 3 different ways and this destroyed the nesting of the ToC to navigate easily.  So, it's a big change. Hopefully it improves things.

There's still room for improvement, but I am much happier with the logical flow and semantics for now.

Oglopf commented:
I also reworded the "Insecure Options" page to call out the reason so we don't have to explain it constantly along with making it a bit more severe in the warning of an insecure prod instance.

moffatsadeghi commented on source/ood-authentication.rst:
I really like that you have titled this OOD Documentation for clearer communication, but the URL needs to stay https://osc.github.io/ood-documentation/latest/authentication.html so that when people are configuring OOD and they get redirected to the documentation, they don't now get a 404 page.  

https://github.com/OSC/ondemand/blob/4acd0a2a3ac3cb46787f757005255d1890be10f2/ood-portal-generator/templates/no-auth.conf.erb#L11

Could you keep the title of the section the same but revert back the URL to the original?

moffatsadeghi replied:
Maybe a more clearer way I could could have put it:

The heading and the text can still have "OOD Authentication" but the URL / maybe the file name needs to keep "Authentication"

johrstrom replied:
>  but the URL needs to stay https://osc.github.io/ood-documentation/latest/authentication.html 

Yes I believe it's imperative we keep the URL for compatibility with previous versions.

Travis, Emily mentioned to me you might need some help with her suggestion to change the URL back to `authentication.html` from `ood-authentication.html`. If that's the case, I can help walk you through the process and why sphinx is throwing errors. Alternatively I'm happy to provide a diff that you can apply or open another PR against this or any other branch.

moffatsadeghi reviewed (changes requested):
This is absolutely awesome, Travis, and I want to get this into our docs soon.  Right now, these changes are in develop, which means they will not get pushed out until 4.1 timeframe because I've confirmed there are several other things waiting in develop that should not go to latest because of how they are tied to future releases.  Is it possible for you to move these changes to latest so we can get it into the docs way sooner? Sorry that I did not mention having you doing this in latest sooner :(
